src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -574,6 +574,34 @@ class Global(var currentSettings: Settings, var reporter: Reporter)
     val runsRightAfter = None
   } with GenBCode
 
+  // phaseName = "analyzeDeps"
+  object analyzeDeps extends {
+    val global: Global.this.type = Global.this
+  } with SubComponent {
+    val phaseName = "analyzeDeps"
+    val runsAfter = List("jvm")
+    val runsRightAfter = None
+
+    def newPhase(prev: Phase): GlobalPhase = {
+      new AnalyzeDepsPhase(prev)
+    }
+    private class AnalyzeDepsPhase(prev: Phase) extends GlobalPhase(prev) {
```

Comment:
What sort of performance do you get with: https://github.com/retronym/classpath-shrinker ?

That only adds 10ms when compiling the standard library, which has a reasonably sized symbol table to walk through.